pkg/planner/core/operator/physicalop/physical_indexlookup_reader.go

```diff
@@ -205,14 +222,51 @@ func (p *PhysicalIndexLookUpReader) ExplainInfo() string {
 }
 
 // Init initializes PhysicalIndexLookUpReader.
-func (p PhysicalIndexLookUpReader) Init(ctx base.PlanContext, offset int) *PhysicalIndexLookUpReader {
+func (p PhysicalIndexLookUpReader) Init(ctx base.PlanContext, offset int, tryPushDownIndexLookUp bool) *PhysicalIndexLookUpReader {
 	p.BasePhysicalPlan = NewBasePhysicalPlan(ctx, plancodec.TypeIndexLookUp, &p, offset)
-	p.TablePlans = FlattenPushDownPlan(p.TablePlan)
-	p.IndexPlans = FlattenPushDownPlan(p.IndexPlan)
 	p.SetSchema(p.TablePlan.Schema())
+	p.SetStats(p.TablePlan.StatsInfo())
+	if tryPushDownIndexLookUp {
+		p.tryPushDownLookUp(ctx)
+	}
+	p.TablePlans = FlattenListOrTiFlashPushDownPlan(p.TablePlan)
+	p.IndexPlans, p.IndexPlansUnNatureOrders = FlattenTreePushDownPlan(p.IndexPlan)
 	return &p
 }
 
+// tryPushDownLookUp tries to push down the index lookup to TiKV.
+func (p *PhysicalIndexLookUpReader) tryPushDownLookUp(ctx base.PlanContext) {
+	intest.Assert(!p.IndexLookUpPushDown)
+	if p.KeepOrder {
```

Comment:
Could it be included in `checkIndexLookUpPushDownSupported`?

Reply:
I think it's hard to do this because keepOrder cannot be determined in `checkIndexLookUpPushDownSupported`. And some logic plan may result in multiple plans, with some keeping order and some not.  In the future, if we support the keep order in the push down case, this checks can be removed.

Reply:
Better to add comments about it.